pkg/rbd/rbd.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2024 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// List retrieves and displays the Ceph block pools and their associated images.
+func ListRbdVolumes(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace string, clusterNamespace string) {
+	blockPoolList, err := clientsets.Rook.CephV1().CephBlockPools(clusterNamespace).List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("failed to list CephBlockPools: %w", err))
+	}
+	var blockPoolNames map[string]string
+	blockPoolNames = make(map[string]string)
+
+	for _, blockPool := range blockPoolList.Items {
+		blockPoolNames[blockPool.Name] = "--"
+	}
+
+	cmd := "rbd"
```

Comment:
We also need to take care of listing the images in the radosNamespace inside the pool not just in default rados namespace

Reply:
yes thats correct, i think we need to accept the --pool --namespace CR names as the arguments or we might end up listing all the pools and all the rados namespaces. @subhamkrai @travisn WDYT?

Reply:
Correct, we need to use `--namespace` also to get the image namespace

Reply:
```
blockPoolList, err := clientsets.Rook.CephV1().CephBlockPools(clusterNamespace).List(ctx, v1.ListOptions{})
```
Currently, this code is checking for CephBlockPools CRs only within the specified clusterNamespace (rook-ceph). Would you like to extend the check to see if CephBlockPools CRs exist in other namespaces as well?

If so, we can update it to:
```
blockPoolList, err := clientsets.Rook.CephV1().CephBlockPools("").List(ctx, v1.ListOptions{})
```

Using `""` as the namespace parameter makes the List call retrieve resources from all namespaces.

Additionally, would you like to include a column for the image namespace? This could help provide more context, as in the following example:

```
$ ./bin/kubectl-rook-ceph rbd ls
poolName     imageName                                     namespace
--------     ---------                                     ---------
builtin-mgr  --                                           rook-ceph
replicapool  csi-vol-f090b7e1-a821-4a71-a7d2-7e3e2aaea8d1 rook-ceph2
```

Let me know if you think this approach would be useful!

Reply:
We need to get the namespace from the blockpool CR, it is not the kubernetes namespace. Check https://rook.github.io/docs/rook/latest/CRDs/Block-Storage/ceph-block-pool-rados-namespace-crd/?h=radosnamespace#spec  field `remoteNamespace`.

Is that correct @Madhu-1 ?

Reply:
Done 
```
$ ./bin/kubectl-rook-ceph rbd ls
poolName     imageName                                     nameSpace    
--------     ---------                                     ---------    
builtin-mgr  --                                            --           
replicapool  csi-vol-9c1f44c5-4320-4d96-9f83-aa6c16e4522d  namespace-a  
```
I created new CR https://github.com/rook/rook/blob/c5df2fe7214b51b1297b0250d2153970047c6d74/deploy/examples/radosnamespace.yaml#L11-L12

Reply:
@subhamkrai Do I need to test with  this yaml https://github.com/rook/rook/blob/c5df2fe7214b51b1297b0250d2153970047c6d74/deploy/examples/radosnamesapce-mirrored.yaml#L13-L14

Reply:
Yes, better we test that

Reply:
@Madhu-1 @subhamkrai

Reply:
spec.name if its set or else metadata.name https://github.com/rook/rook/blob/c5df2fe7214b51b1297b0250d2153970047c6d74/deploy/examples/radosnamespace.yaml#L4-L9

Reply:
@Madhu-1 @subhamkrai done. I updated `github.com/rook/rook/pkg/apis `